kubernetes/skaffold/src/main/kotlin/com/google/kubernetes/tools/skaffold/SkaffoldExecutorService.kt

```diff
@@ -43,16 +44,22 @@ abstract class SkaffoldExecutorService {
     }
 
     /** Path for Skaffold executable, any form supported by [ProcessBuilder] */
-    protected abstract var skaffoldExecutablePath: Path
+    protected abstract fun getSkaffoldExecutablePath(): Path
 
     fun getSystemPath(): String = System.getenv("PATH")
 
-    fun isSkaffoldAvailable(): Boolean = getSystemPath().split(File.pathSeparator)
-            .asSequence()
-            .map { it + File.separator + "skaffold" } // convert each to a possible path to the executable
-            .any {
-                File(it).exists() && File(it).canExecute()
-            }
+    fun isSkaffoldAvailable(): Boolean {
+        val skaffoldPathOverride = KubernetesSettingsService.instance.skaffoldExecutablePath
+
+        val isSkaffoldOnPath = getSystemPath().split(File.pathSeparator)
+                .asSequence()
+                .map { it + File.separator + "skaffold" }
+                .any {
+                    File(it).exists() && File(it).canExecute()
+                }
+
+        return skaffoldPathOverride.isNotBlank() || isSkaffoldOnPath
```

Comment:
Does it make sense to check skaffoldPathOverride for validity and presence of skaffold executable as well to be 100% sure it works?

Reply:
I think that would be a nice follow up. We currently aren’t checking that the executable on the path works either. How about supplement this in another PR?

Reply:
https://github.com/GoogleCloudPlatform/google-cloud-intellij/issues/2404